In [5]:
from google.colab import drive
drive.mount('/content/drive')
data_path = '/content/drive/MyDrive/chest_xray/'

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


Sto considerando di usare i seguenti modelli di ML:
1. ResNet.
2. GoogLeNet.
3. AlexNet.

In [6]:
import os
import pathlib

data_dir = pathlib.Path(os.path.join(data_path, 'train'))
total = len(list(data_dir.glob('**/*.jpeg')))
bacteria = len(list(data_dir.glob('**/BACTERIA-*.jpeg')))
virus = len(list(data_dir.glob('**/VIRUS-*.jpeg')))
P = len(list(data_dir.glob('**/NORMAL-*.jpeg')))
N = total - P
print(f'#training set = {total}')
print(f'#positives = {P}')
print(f'#negatives = {N} (bacteria = {bacteria}, virus = {virus})')

#training set = 5232
#positives = 1349
#negatives = 3883 (bacteria = 2538, virus = 1345)


Le immagini nel training set sono 5232 di cui: 1349 appartenenti alla classe NORMAL mentre 3883 alla classe PNEUMONIA.
Sto considerando diverse strategie per aumentare la dimensione del dataset:
1. Trasformazioni geometriche: specchiare, ruotare, tagliare, allungare.
   a. Specchiare e ruotare non mi sembrano che possano essere molto utili per questa tipologia di problema. Similmente anche
      il ritagliare l'immagine potrebbe rimuovere porzioni di immagini che potrebbero rivelarsi fondamentali per individuare o meno
      la polmonite.
   b. Tag
2. Modificare la luminosità.

In [ ]:
from PIL import Image
images = list(map(lambda i: Image.open(str(i)), data_dir.glob('**/*.jpeg')))

È necessario ridimensionare le immagini per:
- Utilizzare meno risorse computazionali.
- Velocizzare sia addestramento che inferenza.
Al momento sto considerando le diverse strategie per il ridimensionamento delle immagini:
1. Ridimensionare tutte le immagini a 133x133 dove 133 è l'altezza minima tra tutte le immagini.
2. Ridimensionare tutte le immagini a 290x290 dove 290 è l'altezza media delle immagini